bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -220,17 +220,17 @@ public void catchUp()
       }
 
       if (iterator.hasNext()) {
-        addedData();
+        addedData(false);
       }
     }
 
     logger.debug("Exiting catch up because caughtup = {}", caughtup);
   }
 
   @Override
-  public boolean addedData()
+  public boolean addedData(boolean checkIfListenerHaveDataToSendOnly)
   {
-    if (isReady()) {
+    if (!checkIfListenerHaveDataToSendOnly && isReady()) {
```

Comment:
I don't think that a separate method is necessary. The return value of addedData() means the same in both cases - is there more data to send.